decode.go

```diff
@@ -548,11 +548,8 @@ func resetMap(out reflect.Value) {
 
 func (d *decoder) null(out reflect.Value) bool {
 	if out.CanAddr() {
-		switch out.Kind() {
-		case reflect.Interface, reflect.Ptr, reflect.Map, reflect.Slice:
-			out.Set(reflect.Zero(out.Type()))
-			return true
-		}
+		out.Set(reflect.Zero(out.Type()))
+		return true
```

Comment:
Thank you very much for engaging and trying to fix that issue. Unfortunately, that's likely not what we want, as it changes the behavior and means nulls become valid options for scalars, unlike what we had and intended I believe. I'll have a look at it.